js/core.js

```diff
@@ -600,6 +636,39 @@ var ERMrest = (function (module) {
                 // add to referredBy of the key table
                 foreignKeyRef.key.table.referredBy._push(foreignKeyRef);
             }
+        },
+
+        // returns visible foreignkeys.
+        _visibleForeignKeys: function (context) {
+            var orders = -1;
+            try {
+                var annot = this.annotations.get(module._annotations.VISIBLE_FOREIGN_KEYS);
+                if (context !== undefined && annot && annot.content) {
+                    orders = module._getAnnotationValueByContext(context, annot.content);
+                }
+            } catch (exception) {}
+
+            // no annoation, return all outbound and inbound fks
+            if (orders == -1) {
```

Comment:
Since the annotation doesn't specify anything about inbound or outbound; I am returning all visible fks in `_visibleForeignKeys` function, regardless of their type (inbound, outbound). 

However since I only have implemented inbound related references, when I am using the result of this function, I only check for inbound fks.